src/dmtcp_restart_main.cpp

```diff
@@ -0,0 +1,36 @@
+/****************************************************************************
+ *   Copyright (C) 2006-2013 by Jason Ansel, Kapil Arya, and Gene Cooperman *
+ *   jansel@csail.mit.edu, kapil@ccs.neu.edu, gene@ccs.neu.edu              *
+ *                                                                          *
+ *  This file is part of DMTCP.                                             *
+ *                                                                          *
+ *  DMTCP is free software: you can redistribute it and/or                  *
+ *  modify it under the terms of the GNU Lesser General Public License as   *
+ *  published by the Free Software Foundation, either version 3 of the      *
+ *  License, or (at your option) any later version.                         *
+ *                                                                          *
+ *  DMTCP is distributed in the hope that it will be useful,                *
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of          *
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the           *
+ *  GNU Lesser General Public License for more details.                     *
+ *                                                                          *
+ *  You should have received a copy of the GNU Lesser General Public        *
+ *  License along with DMTCP:dmtcp/src.  If not, see                        *
+ *  <http://www.gnu.org/licenses/>.                                         *
+ ****************************************************************************/
+
+#include "dmtcp_restart.h"
+
+using namespace dmtcp;
+
+#define MTCP_RESTART_BINARY "mtcp_restart"
+
+int
```

Comment:
Some comments would help.  Normally, dmtcp_XXX.cpp is for an executable of the same name.

Here, you're breaking the convention, and choosing dmtcp_restart_main.cpp.  This is confusing.

As an alternative, how about naming this file:  `dmtcp_restart.cpp`  (to keep the convention)

Then, you can rename the current `dmtcp_restart.cpp` into something like `dmtcprestartinternal.cpp` or `libdmtcprestart.cpp` (or else `dmtcp_librestart.cpp`, but I prefer the first version, since it's used for `libdmtcprestart.a`), since you're now going convert all of that code into a library that can be linked in more than one way.  After all, you are building a libdmtcprestart.a, similar to `libdmtcpinternal.a`.